File: WebApp/src/Api/Services/ListService.js
export default class ListService {
    GetListItemsByListId = (async (Id) =>{
        let listItems = undefined;
        await fetch('http://localhost:5000/api/toDoList/'+Id)
            .then(res => res.json())
            .then((data) => {
                listItems = data.items.map((item) => ({id:item.id, listId:item.listId, text:item.text, url:item.url, completed:item.completed}));
            });
        return listItems;    
    });
}

File: WebApi/Services/ListService.py
from WebApi.Models.ListItem import ListItem
from WebApi.Models.ToDoList import ToDoList
from WebApi.Data.ListItemData import ListItemData
import json

class ListService():
    def getListItemsByListId(listId):
        #for each list item with listID = listId, create dictionary and return
        listItemData = ListItemData()
        items = listItemData.getByListId(listId)
        return items

    def insertOrUpdateListItem(listItem):
        item = json.loads(listItem)
        listItemData = ListItemData()
        if not item["_id"]:
            result = listItemData.insert(item)
        else:
            result = listItemData.updateListItem(item)   
        return result

    def deleteListItem(itemId):
        listItemData = ListItemData()
        result = listItemData.deleteByItemId(itemId)
        return result
